gnpy/core/convert.py

```diff
@@ -511,7 +512,10 @@ def parse_excel(input_filename):
         raise ValueError(f'Duplicate city: {all_cities}')
     if any(ln.from_city not in all_cities or
            ln.to_city   not in all_cities for ln in links):
-        raise ValueError(f'Bad link.')
+        for lnk in links:
```

Comment:
This checks these links twice, and the error message is printed to stdout. I think it would be better to put it into the exception instead.

I think it would be OK to just get rid of the initial `any(...)` check, perform the explicit loop as this patch is doing, and upon seeing a first failure:
```py
raise NetworkTopologyError(f"Bad link from {lnk.from_city} to {ln.to_city}')
```

Reply:
But then it stops at the first error, and the user has to rerun the program. I would prefer to have the list of all wrong names when I debug my topologies...

Reply:
I still think that encoding an error in an exception is preferred over directly printing it. That way, the info is available to other users than just those using console. Can you just sweep through the list and file a temporary as needed, and then put that info into the exception?

That said, this is "just" an XLS convertor and the REST API won't work with XLS files (I hope!), so my negative vote is a suggestion, not a request.